universal-application-tool-0.0.1/test/support/TestQuestionBank.java

```diff
@@ -47,6 +38,19 @@
   private Map<QuestionEnum, Question> questionCache = new ConcurrentHashMap<>();
   private AtomicLong nextId = new AtomicLong(1L);
 
+  private ImmutableMap<QuestionType, Question> typeToQuestionMap =
```

Comment:
(blocking) Since this is only used in the method you added, it should just be created and returned there. That helps limit the scope, and  it is only created when it is needed.

Reply:
Resolved with a new unit test for the method in question